structslop.go

```diff
@@ -119,13 +136,59 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			}
 		}
 
+		dtyp := dec.Dst.Nodes[atyp].(*dst.StructType)
+		fields := make([]*dst.Field, 0, len(r.optIdx))
+		dummy := &dst.Field{}
+		for _, f := range dtyp.Fields.List {
+			fields = append(fields, f)
+			if len(f.Names) > 1 {
+				for range f.Names[1:] {
+					fields = append(fields, dummy)
+				}
+			}
+		}
+		optFields := make([]*dst.Field, 0, len(r.optIdx))
+		for _, i := range r.optIdx {
+			f := fields[i]
+			if f == dummy {
+				continue
+			}
+			optFields = append(optFields, f)
+		}
+		dtyp.Fields.List = optFields
+
+		var suggested bytes.Buffer
+		if err := decorator.Fprint(&suggested, df); err != nil {
+			return
+		}
 		pass.Report(analysis.Diagnostic{
 			Pos:            n.Pos(),
 			End:            n.End(),
 			Message:        msg,
 			SuggestedFixes: nil,
 		})
+		f := pass.Fset.File(n.Pos())
+		fileDiags[f.Name()] = suggested.Bytes()
 	})
+
+	if apply {
```

Comment:
if !apply { return nil, nil }

Reply:
I wonder this, too. If we invert the condition, then we have 2 `return nil, nil`, one inside `if`, one after the `for` loop.